COBOL Programming Course #1 - Getting Started/COBOL Programming Course #1 - Getting Started.md

```diff
@@ -1903,6 +1903,308 @@ Notice that this line tells you to focus on the GROSS-PAY picture clause in orde
+        10  INSTRUCTOR-ID             PIC 9(10).
+        10  ASSIGNMENT-DETAILS OCCURS 8 TIMES.
+            15  ASSIGNMENT-NAME       PIC X(32).
+            15  ASSIGMMENT-WEIGHTAGE  PIC 9(03).
+```
+
+Here, we are defining a degree program which have 10 course and each course will have 8 assignments. What if we don't know beforehand how many times will a table element occurs? To solve that, we can use variable-length table, using the OCCURS DEPENDING ON (ODO) clause which we will be going into more details on a later section.
+
+## Referring to an item in a table
+
+While a table element have a collective name, the individual items within does not have a unique name. To refer to an item, we can either use subscript, index, or a combination of both.
+
+### Subscripting
+
+Subscripting is using the data name of the table element, along with its occurence number (which is called a subscript). The lowest possible subscript number is 1, which defines the first occurence of a table element. We can also use literal or data name as a subscript. Note that if you are using data name, it must be an elementary numeric integer.
+
+```
+01  TABLE-NAME.
+    05  TABLE-ELEMENT OCCURS 3 TIMES    PIC X(03) VALUE "ABC".
+...
+    MOVE "DEF" TO TABLE-ELEMENT (2)
+```
+
+In the above example, the second table-element will contain "DEF" instead of "ABC".
+
+### Indexing
+
+Alternatively, we can create an index using the INDEXED BY phrase of the OCCURS clause. This index is added to the address of the table to locate an item (as a displacement from the start of the table). For example,
+
+```
+05  TABLE-ELEMENT OCCURS 10 TIMES INDEXED BY INX-A   PIC X(03).
+```
+
+Here, INX-A is an index name. The compiler will calculate the value in the index as the occurence number minus 1 multiplied by the length of the table element. So, for example, for the second occurence of TABLE-ELEMENT, the binary value contained in INX-A is (2-1) * 3, or 3.
+
+If you happen to have another table with the same number of table elements of the same length, you can use an index name as a reference for both table.
+
+We can also define an index data item using the USAGE IS INDEX clause. These index data items can be used with any table. For example,
+
+```
+77  INX-B  USAGE IS INDEX.
+...
+    SET INX-A TO 10.
+    SET INX-B TO INX-A.
+    PERFORM VARYING INX-A FROM 1 BY 1 UNTIL INX-A > INX-B
+        DISPLAY TABLE-ELEMENT (INX-A)
+        ...
+    END-PERFORM.
+```
+
+The index name INX-A is used to traverse the TABLE-ELEMENT table, while INX-B is used to hold the index of the last element of the table. By doing this, we minimize the calculation of offsets and no conversion will be necessary for the UNTIL condition.
+
+We can also increment or decrement an index name by an elementary integer data item. For example,
+
+```
+SET INX-A DOWN BY 3
+```
+
+The integer there represents the number of occurences. So it will be converted to an index value first before it adds or subtract the index.
```

Comment:
This section feels a little dense... perhaps an example of when you would use this would help